src/twisted/internet/defer.py

```diff
@@ -53,6 +53,13 @@ class TimeoutError(Exception):
 
 
 
+class NoCurrentResult(BaseException):
```

Comment:
I think it'd be good if users could find it without diving into the source code.  That could still just be part of a docstring (and maybe `extractCurrentResult`\ 's docstring if that makes more sense) since those go into the generate api docs served on the website.

If it doesn't easily fit into the API docs, though, I'd be satisfied with a comment.  At least then the info is recorded and someone might be able to improve the doc situation later on.